ansible-cli.js

```diff
@@ -0,0 +1,195 @@
+const { promisify } = require("util");
+const childProcess = require("child_process");
+const _ = require("lodash");
+const {
+  prependNamedArguments,
+  createDockerVolumeConfig,
+  extractMountPointsFromVolumeConfigs,
+  mergeVolumeConfigsEnvironmentVariables,
+  validatePaths,
+  createDockerVolumesString,
+  createEnvironmentVariablesString,
+} = require("./helpers");
+const { ANSIBLE_DOCKER_IMAGE } = require("./consts.json");
+
+const exec = promisify(childProcess.exec);
+
+async function execute({
+  config: unparsedAnsibleConfig,
+  command,
+}) {
+  const {
+    volumeConfigs,
+    environmentVariables,
+    ansibleConfig,
+  } = await parseAnsibleConfig(unparsedAnsibleConfig);
+
+  const ansibleCommand = createAnsibleCommand(command, ansibleConfig);
+  const sanitizedAnsibleCommand = sanitizeCommand(ansibleCommand);
+  const dockerCommand = createDockerCommand(sanitizedAnsibleCommand, {
+    environmentVariables: Object.keys(environmentVariables),
+    volumeConfigs,
+  });
+
+  let result;
+  try {
+    result = await exec(dockerCommand, {
+      env: environmentVariables,
+    });
+  } catch (error) {
+    throw new Error(error.stdout || error.stderr || error.message || error);
+  }
+
+  return result;
+}
+
+async function parseAnsibleConfig(rawAnsibleConfig) {
+  let environmentVariables = {};
+  const volumeConfigs = [];
+  const ansibleConfig = rawAnsibleConfig;
```

Comment:
In whole this function you're mixing `ansibleConfig` and `rawAnsibleConfig`. Pick which is the information source and which is the result and then use former to read the data and latter to save it and return it.